src/StakeVault.sol

```diff
@@ -42,6 +44,13 @@ contract StakeVault is Ownable {
         _;
     }
 
+    modifier onlyTrustedStakeManager() {
+        if (!_stakeManagerImplementationTrusted()) {
+            revert StakeVault__StakeManagerImplementationNotTrusted();
+        }
+        _;
+    }
```

Comment:
This modifier prevents users to stake/unstake/lock etc as soon as an upgrade on the stake manager has happened.
We basically force them to either leave or accept the upgrade by explicitly trusting the new stake manager (will be done in a future commit).